python/cuml/test/test_umap.py

```diff
@@ -84,47 +119,99 @@ def test_umap_trustworthiness_on_iris():
     assert trust >= 0.964
 
 
-def test_umap_trustworthiness_on_iris_random_init():
-    iris = datasets.load_iris()
-    data = iris.data
-    embedding = UMAP(
-        n_neighbors=10, min_dist=0.01,  init="random"
-    ).fit_transform(data)
-    trust = trustworthiness(iris.data, embedding, 10)
-    assert trust >= 0.95
+@pytest.mark.parametrize('name', dataset_names)
+def test_umap_fit_transform_trust(name):
 
+    if name == 'iris':
+        iris = datasets.load_iris()
+        data = iris.data
+        labels = iris.target
 
-@pytest.mark.parametrize('should_downcast', [True, False])
+    elif name == 'digits':
+        digits = datasets.load_digits(n_class=5)
+        data = digits.data
+        labels = digits.target
+
+    elif name == 'wine':
+        wine = datasets.load_wine()
+        data = wine.data
+        labels = wine.target
+    else:
+        data, labels = make_blobs(n_samples=5000, n_features=10,
+                                  centers=10, random_state=42)
+
+    model = umap.UMAP(n_neighbors=10, min_dist=0.01)
+    cuml_model = UMAP_cuml(n_neighbors=10, min_dist=0.01, verbose=False)
+    embedding = model.fit_transform(data)
+    cuml_embedding = cuml_model.fit_transform(data)
+
+    trust = trustworthiness(data, embedding, 10)
+    cuml_trust = trustworthiness(data, cuml_embedding, 10)
+
+    assert array_equal(trust, cuml_trust, 1e-1, with_sign=True)
+
+
+@pytest.mark.parametrize('nrows', [pytest.param(20, marks=pytest.mark.unit),
+                                   pytest.param(100000,
+                                                marks=pytest.mark.stress),
+                                   pytest.param(5000,
+                                                marks=pytest.mark.quality)])
+@pytest.mark.parametrize('n_feats', [pytest.param(10, marks=pytest.mark.unit),
+                                     pytest.param(1000,
+                                                  marks=pytest.mark.stress),
+                                     pytest.param(500,
+                                                  marks=pytest.mark.quality)])
+@pytest.mark.parametrize('should_downcast', [True])
 @pytest.mark.parametrize('input_type', ['dataframe', 'ndarray'])
-def test_umap_data_formats(input_type, should_downcast):
+def test_umap_data_formats(input_type, should_downcast,
+                           nrows, n_feats):
 
     dtype = np.float32 if not should_downcast else np.float64
+    n_samples = nrows
+    n_feats = n_feats
+    if nrows > 1000:
+        X, y = datasets.make_blobs(n_samples=n_samples*10,
+                                   n_features=n_feats, random_state=0)
+    else:
+        # For now, FAISS based nearest_neighbors only supports single precision
+        digits = datasets.load_digits(n_class=9)
```

Comment:
It seems confusing that nrows = 20 really means 'load digits and use all of that'... do we need to use digits? Is there a clearer way to show this, like: ```
  if isinstance(data_set, str) and data_set == 'digits':
     # use digits
  else:
     # use blobs with nrows = data_set
```
 else: